### PR TITLE
Zoltan2:  code cleanup and remove GlobalMPISession

### PR DESCRIPTION
**Pull request by kddevin**
@trilinos/zoltan2 

## Motivation and Context

General code cleanup:
* Removed remaining Teuchos::GlobalMPISession after merge with #3145 
* Replaced most calls to Teuchos::DefaultComm with Tpetra::getDefaultComm; only call to Teuchos for default SerialComm remains
* Removed using namespace std
* Some variable name changes for consistency across tests

## How Has This Been Tested?

mac with clang; linux with gnu

## Checklist

- [ x] My commit messages mention the appropriate GitHub issue numbers.
- [ x] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.
- [ x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

**Comment by mhoemmen** [comment]
The best thing you did was removing `using namespace std;` at global scope :D  Thanks!!!

File: packages/zoltan2/example/block/kokkosBlock.cpp [hunk 1]
```diff
@@ -64,11 +62,10 @@ using Teuchos::RCP;
  *  \todo write some examples that don't use teuchos
  */
 
-int main(int argc, char *argv[]) {
-  Teuchos::GlobalMPISession mpiSession(&argc, &argv);
-  Kokkos::initialize(argc, argv);
+int main(int narg, char *arg[]) {
+  Tpetra::ScopeGuard tscope(&narg, &arg);
+  Teuchos::RCP<const Teuchos::Comm<int> > comm = Tpetra::getDefaultComm();
```
Comment (mhoemmen) [inline packages/zoltan2/example/block/kokkosBlock.cpp]: It looks like the corresponding `Kokkos::finalize` is still at the bottom of the file.  ScopeGuard will be grumpy about the double finalize.  Unfortunately, Kokkos only has an `is_initialized()` function, not an `is_finalized()` function.  (That's different than MPI, alas.)  This means that Tpetra can't check whether the user made a mistake like this.

File: packages/zoltan2/example/geometric/rcb_C.cpp [hunk 2]
```diff
@@ -186,14 +185,14 @@ int main(int argc, char *argv[])
 
   // Create a Zoltan2 input adapter that includes weights.
 
-  vector<const scalar_t *>coordVec(2);
-  vector<int> coordStrides(2);
+  std::vector<const scalar_t *>coordVec(2);
```
Comment (mhoemmen) [inline packages/zoltan2/example/geometric/rcb_C.cpp]: It's OK to have `using namespace std;` at function scope, just not at global scope.  Also, you still have `using std::vector` above (which is fine, it's just that you didn't need to make this change in that case).

**Comment by kddevin** [comment]
Good catch on  Kokkos::finalize(), @mhoemmen .  Thanks.
Both suggestions are fixed

**Comment by mhoemmen** [comment]
Thanks @kddevin ! :D

**Pull request merged**